src/allocator.rs

```diff
@@ -232,6 +232,10 @@ impl Allocator {
         Ok(-(self.atom_vec.len() as i32))
     }
 
+    pub fn atom_eq(&self, lhs: NodePtr, rhs: NodePtr) -> bool {
+        self.atom(lhs) == self.atom(rhs)
```

Comment:
this is the new function. The idea is to give the allocator an opportunity (in the future) lazily convert atoms to bytes. This function is another opportunity to keep atoms in their native/unserialized form